docs/config.md

```diff
@@ -73,7 +73,7 @@ setting        | mandatory | values                                        | def
 key            |     Y     | string                                        | N/A     |
 type           |     Y     | sendAllMatch/sendFirstMatch/consistentHashing | N/A     | send to all destinations vs first matching destination vs distribute via consistent hashing
 prefix         |     N     | string                                        | ""      |
-sub            |     N     | string                                        | ""      |
+substr         |     N     | string                                        | ""      |
```

Comment:
huh?
this token is declared in imperatives.go
```
{Token: optSub, Pattern: "sub="},
```

Reply:
hmm right, this seems to be a discrepancy with the config file parsing then.... 

because this route in the config file:
```
[[route]]
key = 'carbon-default'
type = 'sendFirstMatch'
sub = 'testvalue'
destinations = [
  '127.0.0.1:2003 pickle=false'
]
```
gets printed like this when crng starts
```
## Routes:
  type            key             prefix  substr  regex
============================================================================================
> sendFirstMatch  carbon-default
  prefix  substr  regex  addr              spoolDir                    spool  pickle  online
  ------------------------------------------------------------------------------------------
                         127.0.0.1:2003    /var/spool/carbon-relay-ng  false  false   false
````

while this route in the config file:
```
[[route]]
key = 'carbon-default'
type = 'sendFirstMatch'
substr = 'testvalue'
destinations = [
  '127.0.0.1:2003 pickle=false'
]
```
gets printed like this:
````
## Routes:
  type            key             prefix  substr     regex
============================================================================================
> sendFirstMatch  carbon-default          testvalue
  prefix  substr  regex  addr              spoolDir                    spool  pickle  online
  ------------------------------------------------------------------------------------------
                         127.0.0.1:2003    /var/spool/carbon-relay-ng  false  false   false
````

Reply:
i think we should consistently use the string `sub` everywhere, not `substr`, because it is shorter and it's still quite obvious what it is.

Reply:
ok

Reply:
What's the best way to change this from `sub` to `substr`? Should I make it backwards compatible, so it parses both correctly while giving `sub` preference in case both are defined? Or should I make a breaking change where `substr` doesn't get parsed anymore? I'd lean towards the former, otherwise we just annoy users unnecessarily, and implementing backwards compatibility would be very simple.

Reply:
backwards compatibility is important.
BUT!
> i think we should consistently use the string sub everywhere, not substr,
vs
> What's the best way to change this from sub to substr? 

?

Reply:
sorry, i wrote it the wrong way around, i meant `What's the best way to change this from substr to sub?`